routing_model/index.html

```diff
@@ -433,37 +439,181 @@ <h2>Process for Adult/Elder promotion and demotion</h2>
+      style ProcessElderChange fill:#f9f,stroke:#333,stroke-width:4px
+      ProcessElderChange -->RestartTimeout
 
       CheckElderChange -- "No<br/>changes" --> RestartTimeout
 
+      RestartTimeout["schedule(<br/>CheckElderTimeout)"]
+      RestartTimeout-->LoopEnd
+
+      CheckMergeNeeded --"merge_needed()<br/>or<br/>has_merge_infos()"-->Concurrent1
+
+
+      Concurrent1{"Concurrent<br/>paths"}
+      Concurrent1 --> ProcessMerge
+      Concurrent1 --> LoopEnd
+
+      ProcessMerge["ProcessMerge"]
+      style ProcessMerge fill:#f9f,stroke:#333,stroke-width:4px
+      ProcessMerge --> RestartTimeout
+
+      LoopEnd --> LoopStart
+  </div>
+
+  <h2>Process Adult/Elder promotion and demotion needed from last check</h2>
+  <p>Vote for Add for new elders, Remove for no longer elders and NewSectionInfo<br/>
+     This handle any change, it does not care whether one or all elder are changed, this is decided by the calling function.
+ </p>
+  <div class="mermaid">
+      graph TB
+      ProcessElderChange["ProcessElderChange<br/>(Take elder changes)"]
+      style ProcessElderChange fill:#f9f,stroke:#333,stroke-width:4px
+
+      EndRoutine["End of ProcessElderChange<br/>(shared state)"]
+      style EndRoutine fill:#f9f,stroke:#333,stroke-width:4px
+
+      ProcessElderChange -->  MarkAndVoteSwapNewElder
+      MarkAndVoteSwapNewElder["vote_for(Add) for new elders<br/>vote_for(Remove) for now adults nodes<br/>vote_for(NewSectionInfo)<br/><br/>WAITED_VOTES.insert(all votes)"]
+      MarkAndVoteSwapNewElder --> LoopStart
+
+      WaitFor(("Wait for 5:"))
+      LoopStart --> WaitFor
+
+      Consensus((Consensus))
+      WaitFor-- Parsec<br/>consensus --> Consensus
+
       Consensus -- "WAITED_VOTES.contains(vote)" --> OneVoteConsensused
       OneVoteConsensused["WAITED_VOTES.remove(vote)"]
 
       OneVoteConsensused --> WaitComplete
       WaitComplete(("Check?"))
-      WaitComplete--"WAITED_VOTES.is_empty()<br/>(Wait complete)"-->MarkNewElderAdults
+      WaitComplete--"WAITED_VOTES<br/>.is_empty()<br/>(Wait complete)"-->MarkNewElderAdults
       MarkNewElderAdults["For all nodes:<br/>set_elder_status(node, is_elder)"]
-      MarkNewElderAdults--> RestartTimeout
-      RestartTimeout["schedule(<br/>CheckElderTimeout)"]
-      RestartTimeout-->LoopEnd
-      WaitComplete--"!WAITED_VOTES.is_empty()<br/>(Wait not complete)"--> LoopEnd
+      MarkNewElderAdults--> EndRoutine
+      WaitComplete--"!WAITED_VOTES<br/>.is_empty()<br/>(Wait not complete)"--> LoopEnd
+
+      LoopEnd --> LoopStart
+  </div>
+
+  <h2>ProcessMerge Sub-routine</h2>
+  <p>Vote for ParsecOurMerge, and take over handling any ParsecNeighbourMerge.<br/>
+     Complete when one merge has completed, and a NewSectionInfo is consensused.<br/>
+     If Multi stage merged are require, they will require calling this function again<br/>
+     <br/>
+     store_merge_infos: Store the neighbour merge info, may not be sibling in case of multi merge<br/>
```

Comment:
These will look nicer in collapsible buttons, but don't worry: I can take care of it when I get there.